src/Http/Controllers/PageController.php

```diff
@@ -13,6 +13,11 @@ class PageController extends BaseController
 {
     public function __invoke(Request $request, Page $page, ?Page $childPage = null, Page $grandChildPage = null)
     {
+        // If location based child is only child of parent, redirect to parent alone!!
+        if ($childPage && $childPage->location_based && $page->children->count() === 1) {
+            return redirect(url($page->path));
+        }
```

Comment:
I ended up using the existing `location_based` flag as the trigger for whether or not a child page that is the only child of a parent should redirect to the parent alone.  This won't fully work until the locations package is updated so it sets this flag to true when creating/updating pages for markets and locations.

Reply:
I would prefer renaming that field.

Reply:
Thats fine, what do you want to call it?   For new name, keep in mind that you can't honor the flag unless the only child condition is also met.

Reply:
So this flag will be used for both Market & Location in https://github.com/tipoff/locations/pull/112

I had originally intended this to be used for location-based pages per issue #54 and https://github.com/tipoff/locations/issues/85 but that is no longer needed since we're handling those as a fixed route name segment in the packages.

Let's just use `is_location` for this boolean to denote if the page is a Market or Location. I don't like the name that much since it can also be a market, but I don't want to use something like `is_physical` or some 3rd word for it.

Reply:
I'm going to leave it the same for now. I want to push through this PR and see it in action.